Server/APIs/RoomsAPI.py

```diff
@@ -0,0 +1,110 @@
+    rooms_as_dicts = {}
+    index = 1
+    rooms_as_objects = get_available_rooms_as_objects()
+    for room_as_object in rooms_as_objects:
+        rooms_as_dicts[index] = room_as_object.to_dict()
+        index += 1
+    return jsonify(rooms_as_dicts)
+
+
+@rooms_api.route("/rooms/<room_name>", methods=['GET'])
+@token_required
+def get_room(current_player, room_name):
+    room_as_dict = DBMethods.get_room_details_from_db(room_name)
+    if room_as_dict is None:
+        room_as_dict = {}
+        return jsonify(room_as_dict)
+    else:
+        room_as_object = create_room_object_from_dict(room_as_dict)
+        return jsonify(room_as_object.to_dict())
+
+
+@rooms_api.route("/rooms", methods=['POST'])
+@token_required
+def host_room(current_player):
+    data = request.json
+    room_name = data['room_name']
+    if DBMethods.check_if_room_with_same_name_already_exists(room_name):
+        return RoomsAPIErrors.ROOM_NAME_OCCUPIED.value
+    if DBMethods.check_if_player_already_in_a_room(current_player.username):
+        return RoomsAPIErrors.PLAYER_ALREADY_IN_A_ROOM.value
+    room_id = str(uuid.uuid4())
+    room = Room(room_id, room_name, RoomState.WAITING.value, [current_player], current_player)
+    DBMethods.add_room_to_db(room)
+    DBMethods.update_player_linked_room_in_db(current_player.username, room_id)
+    return "Your room has been created successfully"
+
+
+@rooms_api.route("/rooms/<room_name>", methods=['PUT'])
+@token_required
+def join_room(current_player, room_name):
+    room_name_from_player_input = room_name
+    if DBMethods.check_if_player_already_in_a_room(current_player.username):
+        return RoomsAPIErrors.PLAYER_ALREADY_IN_A_ROOM.value
+    rooms = get_available_rooms_as_objects()
+    for room in rooms:
+        if room.name == room_name_from_player_input:
+            if room.state == RoomState.PLAYING.value:
+                return RoomsAPIErrors.ROOM_IN_PLAYING_STATE.value
+            DBMethods.update_player_linked_room_in_db(current_player.username, room.id)
+            return f"You have successfully joined room {room_name_from_player_input}"
+    return RoomsAPIErrors.ROOM_NAME_DOESNT_EXIST.value
+
+
+# @rooms_api.route("/rooms/<room_name>", methods=['PUT'])
+# def leave_room():
+#     data = request.json
+#     player = Player.create_player_object_from_dict(data)
+#     if not DBMethods.check_if_player_already_in_a_room(player.id):
+#         return "You are not in any room"
+#     DBMethods.update_player_linked_room_in_db(player.id, None)
+#     return "You have successfully left the room"
+
+
+def get_available_rooms_as_objects():
+    rooms_as_dicts = DBMethods.get_available_rooms_from_db()
+    print(rooms_as_dicts)
+    rooms_as_objects = []
+    for room_as_dict in rooms_as_dicts:
+        room_as_object = create_room_object_from_dict(room_as_dict)
+        rooms_as_objects.append(room_as_object)
+    return rooms_as_objects
+
+
+def create_room_object_from_dict(room_as_dict):
```

Comment:
You have this logic already in the client, and in other places. 
I suggest moving them to the room and player for having them in only one place.